modules/basics/src/main/java/com/opengamma/basics/date/DayCount.java

```diff
@@ -79,15 +80,52 @@ public default double yearFraction(LocalDate firstDate, LocalDate secondDate) {
    * Gets the year fraction between the specified dates.
    * <p>
    * Given two dates, this method returns the fraction of a year between these
-   * dates according to the convention.
+   * dates according to the convention. The dates must be in order.
    * 
    * @param firstDate  the first date
    * @param secondDate  the second date, on or after the first date
    * @param scheduleInfo  the schedule information
-   * @return the year fraction
+   * @return the year fraction, zero or greater
+   * @throws IllegalArgumentException if the dates are not in order
+   * @throws UnsupportedOperationException if the year fraction cannot be obtained
+   */
+  public abstract double yearFraction(LocalDate firstDate, LocalDate secondDate, ScheduleInfo scheduleInfo);
+
+  /**
+   * Gets the relative year fraction between the specified dates.
+   * <p>
+   * Given two dates, this method returns the fraction of a year between these
+   * dates according to the convention. If the first date is after the second date,
+   * the result is calculated by swapping the dates to be in order and negating the result.
+   * <p>
+   * This uses a simple {@link ScheduleInfo} which has the end-of-month convention
+   * set to false, but throws an exception for other methods.
+   * Certain implementations of {@code DayCount} need the missing information,
+   * and thus will throw an exception.
+   * 
+   * @param firstDate  the first date
+   * @param secondDate  the second date, which may be before the first date
+   * @return the year fraction, may be negative
+   * @throws UnsupportedOperationException if the year fraction cannot be obtained
+   */
+  public default double relativeYearFraction(LocalDate firstDate, LocalDate secondDate) {
+    return relativeYearFraction(firstDate, secondDate, DayCounts.SIMPLE_SCHEDULE_INFO);
+  }
+
+  /**
+   * Gets the relative year fraction between the specified dates.
+   * <p>
+   * Given two dates, this method returns the fraction of a year between these
+   * dates according to the convention. If the first date is after the second date,
+   * the result is calculated by swapping the dates to be in order and negating the result.
+   * 
+   * @param firstDate  the first date
+   * @param secondDate  the second date, which may be before the first date
+   * @param scheduleInfo  the schedule information
+   * @return the year fraction, may be negative
    * @throws UnsupportedOperationException if the year fraction cannot be obtained
    */
-  public double yearFraction(LocalDate firstDate, LocalDate secondDate, ScheduleInfo scheduleInfo);
+  public abstract double relativeYearFraction(LocalDate firstDate, LocalDate secondDate, ScheduleInfo scheduleInfo);
```

Comment:
Given what the javadoc implies, why isn't this a default implementation? i.e. calling ``yearFraction()``?